documentation/add/07-sqw_redesign.md

```diff
@@ -1,58 +1,84 @@
 
-The SQW object exists in two distinct forms: 
+The `sqw` object contains two information blocks, describing experiment from different sides and under different approximations:
+
+- `dnd-object` which is the part of the `SQW` object contains processed neutron image data, expressed in the system of coordinates, which has a physical meaning. Very often its `hkl-dE` coordinate system, but it is possible to define other coordinate system of interest too.
+- `sqw-object` itself, in addition to the image data, contains the records of all neutron events occurred in the experiments and expressed (currently) in Crystal Cartesian coordinate system, the informations about all experimental runs the neutron events have been obtained from, information about the instrument and sample used in experiment, the instrument states during experiment and the information about detector pixel array, used to collect neutron events. This information is sufficient to calculate resolution function of the instrument. The neutron events are held in the `PixelData` class, and arranged in such a way, that selection of a part of the `dnd-object` neutron image allows the effective selection of the appropriate part of the `PixelData` contributed into this part of image.
+
 
-- "SQW" which contains all experimental data and the detector pixel array,
-- "DND" which contains only processed neutron image data
+The two objects are treated by Horace as 'first class citizens'. The objects share a common API where that is appropriate. In an sqw object, operations are performed on the pixel array data and the `dnd` image data recalculated from this. For `dnd` objects the operations are performed directly on the image data.
+In addition, the extra information and raw data contained on the `sqw` object allows a scattering model to be fitted.
 
-The two objects are treated by Horace as 'first class citizens'. The objects share a common API where that is appropriate. In an SQW object, operations are performed on the pixel array data and the current image data recalculated from this. For DND objects the operations are performed directly on the image data. 
-In addition, the extra information and raw data contained on the SQW object allows a scattering model to be fitted.
+## Main Classes:
 
-## Main Classes
+The class diagram describing the relationship between main Horace classes is provided on the Fig.1:
 
-![Class Overview](../diagrams/sqw-dnd.png)
+![Fig.1. Class Overview](../diagrams/sqw-dnd.png)
 
 
 
 ### SQWDnDBase
 
-Abstract base class for the SQW and DND objects.
+Abstract base class for the `sqw` and `dnd` objects, describing the interface common for both `sqw` and `dnd` objects.
 
-This includes the image data and main header information, available in both the `SQW` and `DnD` classes.
+May include code of the methods with work identically on both `sqw` and `dnd` objects. .
```

Comment:
I'm going to note here that dndbase and sqw both inherit from this, so that in principle the SQWDnDBase superclass object is duplicated for SQWs in the SQW class itself and in its `data` member. At present this is unimportant as its methods are largely static and its properties constant, but it means that care should be taken if the class is added to in future

Reply:
not entirely sure what you mean by this. Diagrams on Fig1 and Fig 2 show all inheritance path and interfaces available (crudely)